ktor-client/ktor-client-features/ktor-client-logging/jvm/src/io/ktor/client/features/logging/LoggerJvm.kt

```diff
@@ -10,3 +10,45 @@ actual val Logger.Companion.DEFAULT: Logger
             delegate.info(message)
         }
     }
+
+
+/***
+ * Workaround Logger to log long messages that would otherwise be truncated in Android.
+ */
+val Logger.Companion.ANDROID: Logger
+    get() = MaxLengthLogger()
+
+
+class MaxLengthLogger(val MAX_INDEX : Int = 4000) : Logger {
+    private val delegate = LoggerFactory.getLogger(HttpClient::class.java)!!
+    override fun log(message: String) {
+        logLong(message)
+    }
+
+    fun logLong(message: String) {
+
+        val MIN_INDEX = 3000
+
+        // String to be logged is longer than the max...
+        if (message.length > MAX_INDEX) {
+            var theSubstring = message.substring(0, MAX_INDEX)
```

Comment:
`the` suffix makes the code more difficult to read. It would be better to rename.